### PR TITLE
Add make target for mixer codegen

### PR DESCRIPTION
**Pull request by bianpengyuan**
#8543

### CODE REVIEW

File: mixer/docker/Dockerfile.mixer_codegen [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM scratch
+
+ADD mixgen /usr/local/bin/
+
+ENTRYPOINT ["/usr/local/bin/mixgen"]
```
Comment (douglas-reid) [inline mixer/docker/Dockerfile.mixer_codegen]: super nit: add newline.

**Comment by bianpengyuan** [comment]
@douglas-reid Yeah it makes perfect sense to combine protoc and mixer codegen into one image. Let me update the PR to do that.

I don't think we want to include this tool in mixer image though. Essentially it is just a tool which is only used by adapter developer and not needed by mixer workload.

**Comment by bianpengyuan** [comment]
@douglas-reid After thinking a little bit more into it, I think it might not make sense to combine protoc and mixgen together in Istio release since it could introduce a dependency circle.. (API needs protoc to get update, while Istio needs API to build). cc @geeknoid who seems to be maintaining the protoc image.

**Comment by bianpengyuan** [comment]
@douglas-reid ptal?

**Comment by douglas-reid** [comment]
Need to rebase, otherwise LGTM.

**Comment by bianpengyuan** [comment]
@douglas-reid rebased. ptal, thanks!

**Comment by bianpengyuan** [comment]
@douglas-reid  Can I also get a approval? :)

**Comment by douglas-reid** [comment]
@bianpengyuan i don't have OWNERS. that approval must come from elsewhere.

**Comment by bianpengyuan** [comment]
@hklai @geeknoid Can you please take a look?

**Pull request merged**